src/components/side_panel/find_and_replace/find_and_replace.ts

```diff
@@ -72,12 +81,58 @@ export class FindAndReplacePanel extends Component<Props, SpreadsheetChildEnv> {
 
     onMounted(() => this.focusInput());
 
+    useEffect(
+      () => {
+        this.focusButton(this.lastClickedButton);
+      },
+      () => [this.env.model.getters.getActiveSheetId()]
+    );
+
+    useEffect(
+      () => {
+        if (
+          this.state.searchOptions.searchScope === "specificRange" &&
+          !this.state.searchOptions.specificRange
+        ) {
+          this.state.searchOptions.specificRange = this.lastSpecificRange;
+        } else if (
+          this.state.searchOptions.searchScope !== "specificRange" &&
+          this.state.searchOptions.specificRange
+        ) {
+          this.lastSpecificRange = this.state.searchOptions.specificRange;
+          this.state.searchOptions.specificRange = undefined;
+        }
+        this.updateSearch();
+      },
+      () => [this.state.searchOptions.searchScope]
+    );
+
     onWillUnmount(() => {
       this.env.model.dispatch("CLEAR_SEARCH");
       this.env.model.dispatch("SET_FORMULA_VISIBILITY", { show: this.showFormulaState });
     });
   }
 
+  private focusButton(buttonName: string) {
+    setTimeout(() => {
+      const el = this.selectionInputref.el!;
+      if (el) {
+        const buttonElement = this.findAndReplaceRef.el!.querySelector(
+          `button[name=${buttonName}]`
+        ) as HTMLButtonElement;
+        buttonElement.focus();
+      }
+    }, 0);
```

Comment:
That's a hack alright. I won't comment on the `setTimeout`, there's probably a way to do it less hack but without proper focus managment it's not easy.

But did you even test it ? My comment was `when we search something on all sheets, when the search switch sheet the side panel lose focus and the grid takes it`. This is not fixed at all ? Why are we doing stuff on the selecitonInput where we want to search input to be focused ? And why focus the button ??

Start by writign a fixing test when fixing thing. And the fix should aim to make the test pass.

Reply:
I have investigated the issue partially and well it seems that we are trying to fight back against the 'autofocus' of the grid when changing sheet. Upon investigation, it seems that the effect was too radical as it would steal the focus of any focusable element inside spreadsheet (be it the composer, or in this very case, a selection input). I have added a commit that aims to solve this issue, I don't think you will need this hack anymore.